hack/dockerfiles/vendor.Dockerfile

```diff
@@ -1,6 +1,6 @@
 # syntax=docker/dockerfile-upstream:master
 
-ARG GO_VERSION=1.20
+ARG GO_VERSION=1.21
```

Comment:
Oh looks like we forgot to bump this one in https://github.com/moby/buildkit/pull/4302